File: iosApp/DesignSystem/Sources/GoPeakDesignSystem/Theme/GoPeakColors.swift
import SwiftUI

/// GoPeak's bespoke color roles — the SwiftUI counterpart of Android's `GoPeakColors`.
///
/// Each role is a **dynamic** `Color` that resolves light/dark automatically, so there is no
/// separate "dark scheme"/"light scheme" object to select — iOS picks the right value per
/// `userInterfaceStyle`. Values are copied exactly from
/// `core/designSystem/.../theme/GoPeakColors.kt` (light / dark).
public struct GoPeakColors: Sendable {

    /// Expressive orange — the highest-emphasis brand CTA (LargePrimaryButton).
    public let brand = Color(light: 0xFFAC8A, dark: 0xFF6B00)
    public let onBrand = Color(light: 0x351000, dark: 0x572000)

    /// Peach — the primary action color (PrimaryButton, SecondaryButton label).
    public let primary = Color(light: 0xA04100, dark: 0xFFB693)
    public let onPrimary = Color(light: 0xFFFFFF, dark: 0x561F00)

    /// Blue highlight used for accents/metrics. Used as foreground and, at low alpha, as a fill.
    public let accent = Color(light: 0x00639A, dark: 0x9CCAFF)
    public let onAccent = Color(light: 0x0A2E4D, dark: 0x0A2E4D)
    /// `accent` at reduced opacity — background behind accent-colored content.
    public let accentContainer = Color(light: 0xCEE5FF, lightAlpha: 0.5, dark: 0x9CCAFF, darkAlpha: 0.2)

    public let background = Color(light: 0xFFF8F5, dark: 0x1D100A)
    public let onBackground = Color(light: 0x231A14, dark: 0xF8DDD2)

    public let surface = Color(light: 0xFFF8F5, dark: 0x261812)
    public let onSurface = Color(light: 0x1D100A, dark: 0xF8DDD2)

    public let surfaceVariant = Color(light: 0xF0E4DD, dark: 0x41312A)
    public let onSurfaceVariant = Color(light: 0x5A4136, dark: 0xE2BFB0)

    public let outline = Color(light: 0xB8A196, dark: 0x5A4136)

    public init() {}
}

File: core/designSystem/src/main/kotlin/com/danielyan/gopeak/designsystem/theme/GoPeakColors.kt
package com.danielyan.gopeak.designsystem.theme

import androidx.compose.runtime.Immutable
import androidx.compose.ui.graphics.Color

/**
 * GoPeak's bespoke color roles. Deliberately NOT a Material 3 [androidx.compose.material3.ColorScheme]
 * — the app's palette uses custom roles (`brand`, `primary`, `accent`) that don't map cleanly to
 * Material semantics. Access via [GoPeakTheme.colors].
 */
@Immutable
data class GoPeakColors(
    /** Expressive orange — the highest-emphasis brand CTA (LargePrimaryButton). */
    val brand: Color,
    val onBrand: Color,
    /** Peach — the primary action color (PrimaryButton, SecondaryButton label). */
    val primary: Color,
    val onPrimary: Color,
    /** Blue highlight used for accents/metrics. Used as foreground and, at low alpha, as a fill. */
    val accent: Color,
    val onAccent: Color,
    /** [accent] at reduced opacity — background behind [accent]-colored content. */
    val accentContainer: Color,
    val background: Color,
    val onBackground: Color,
    val surface: Color,
    val onSurface: Color,
    val surfaceVariant: Color,
    val onSurfaceVariant: Color,
    val outline: Color,
    val isDark: Boolean,
)

val GoPeakDarkColors = GoPeakColors(
    brand = Color(0xFFFF6B00),
    onBrand = Color(0xFF572000),
    primary = Color(0xFFFFB693),
    onPrimary = Color(0xFF561F00),
    accent = Color(0xFF9CCAFF),
    onAccent = Color(0xFF0A2E4D),
    accentContainer = Color(0xFF9CCAFF).copy(alpha = 0.2f),
    background = Color(0xFF1D100A),
    onBackground = Color(0xFFF8DDD2),
    surface = Color(0xFF261812),
    onSurface = Color(0xFFF8DDD2),
    surfaceVariant = Color(0xFF41312A),
    onSurfaceVariant = Color(0xFFE2BFB0),
    outline = Color(0xFF5A4136),
    isDark = true,
)

// Light neutrals are derived (the Figma light frame was not reachable); `primary` is exact.
val GoPeakLightColors = GoPeakColors(
    brand = Color(0xFFFFAC8A),
    onBrand = Color(0xFF351000),
    primary = Color(0xFFA04100),
    onPrimary = Color(0xFFFFFFFF),
    accent = Color(0xFF00639A),
    onAccent = Color(0xFF0A2E4D),
    accentContainer = Color(0xFFCEE5FF).copy(alpha = 0.5f),
    background = Color(0xFFFFF8F5),
    onBackground = Color(0xFF231A14),
    surface = Color(0xFFFFF8F5),
    onSurface = Color(0xFF1D100A),
    surfaceVariant = Color(0xFFF0E4DD),
    onSurfaceVariant = Color(0xFF5A4136),
    outline = Color(0xFFB8A196),
    isDark = false,
)
